collect_app/src/test/java/org/odk/collect/android/preferences/JsonPreferencesGeneratorTest.kt

```diff
@@ -114,6 +118,46 @@ class JsonPreferencesGeneratorTest {
         verifyJsonContent(jsonPrefs, emptyMap<String, Any>(), emptyMap<String, Any>())
     }
 
+    @Test
+    fun `Only preferences from the current project should be included in json`() {
+        // Setup settings for the demo project
+        val generalPrefsForDemoProject = buildMap<String, Any> {
+            put(GeneralKeys.KEY_DELETE_AFTER_SEND, true)
+            put(GeneralKeys.KEY_APP_THEME, "dark_theme")
+        }
+
+        val adminPrefsForDemoProject = buildMap<String, Any> {
+            put(AdminKeys.KEY_GET_BLANK, false)
+            put(AdminKeys.KEY_DELETE_SAVED, false)
+        }
+
+        settingsProvider.getGeneralSettings().saveAll(generalPrefsForDemoProject)
```

Comment:
I'm thinking it might be simpler for `JsonPreferencesGenerator` to take general settings and admin settings in at the constructor rather than a `SettingsProvider` this test class can just create new settings objects (they don't need to be the real ones) and I'm sure you'd need this test.

Reply:
Yes but I like more providing `SettingsProvider` because then even if we make a mistake and for example make `JsonPreferencesGenerator` a singleton (it's not necessary a mistake it might be a feature) then it still would work because `JsonPreferencesGenerator` will take proper settings (for the current project, not settings for the which was the current one when the object was generated)

Reply:
Yeah, I take you point. I think it still feels more complex than it needs to be, but it isn't a big deal and as you say could be useful in the future, so I'm happy for it to stick like this.